controllers/keystoneservice_controller.go

```diff
@@ -170,6 +170,15 @@ func (r *KeystoneServiceReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{}, err
 	}
 
+	// If both the endpoint and the KeystoneAPI is deleted then we can skip
+	// the cleanup of the endpoint in the DB as the DB is going away as well.
+	// Moreover if KeystoneAPI is being deleted then we cannot talk to the
+	// keystone REST API any more. This happens for example during namespace
+	// deletion.
```

Comment:
Maybe the comment here should speak of `service` instead of `endpoint`?

Reply:
yepp. Copy-paste :(